### ISSUE TITLE
'assets' KeyError when running `rgc.list_assets_by_genome()` with post_update hook

### ISSUE THREAD

**Issue by KevinMenden** [databio/refgenie_nfcore#2]
Hi,

I'm trying to use the `post_update` hook to generate a `nextflow.config` file with the refgenie genomes in it, basically copying the code from this repo here to a large extent.

I run into a KeyError for 'assets' in the `rgc.list_assets_by_genome()`, which is the first function called, see here:
https://github.com/KevinMenden/tools/blob/refgenie/nf_core/refgenie.py

I can reproduce the error as follows:
* fresh refgenie `genomes_config.yaml` (`refgenie init -c genomes_config.yaml`)
* `refgenie pull dm6/fasta`

After running this code, the `genomes_config.yaml` looks like this:
```yaml
config_version: 0.4
genome_folder: /home/kevin/qbic/nfcore/refgenie_test
genome_servers: 
 - http://refgenomes.databio.org
genomes:
  8baf9d24ad8f5678f0fe1f5b21a812d410755d49e3123158:
    aliases: 
     - dm6

```
Running `refgenie pull dm6/fasta` a second time after that works.

Any help would be appreciated! 

Cheers,
Kevin

Here's the full traceback:
```python
refgenie pull dm6/fasta
Compatible refgenieserver instances: ['http://refgenomes.databio.org']
No local digest for genome alias: dm6
Setting 'dm6' identity with server: http://refgenomes.databio.org/v3/genomes/genome_digest/dm6
Determined digest for local 'dm6' alias: 8baf9d24ad8f5678f0fe1f5b21a812d410755d49e3123158
Set genome alias (8baf9d24ad8f5678f0fe1f5b21a812d410755d49e3123158: dm6)
Traceback (most recent call last):
  File "/home/kevin/anaconda3/envs/refgenie/lib/python3.9/site-packages/refgenconf/refgenconf.py", line 1482, in pull
    genome = self.get_genome_alias_digest(alias=alias)
  File "/home/kevin/anaconda3/envs/refgenie/lib/python3.9/site-packages/refgenconf/refgenconf.py", line 1705, in get_genome_alias_digest
    return self[CFG_GENOMES_KEY].get_key(alias=alias)
  File "/home/kevin/anaconda3/envs/refgenie/lib/python3.9/site-packages/yacman/alias.py", line 223, in get_key
    raise UndefinedAliasError("No key defined for: {}".format(alias))
yacman.exceptions.UndefinedAliasError: No key defined for: dm6

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kevin/anaconda3/envs/refgenie/lib/python3.9/site-packages/attmap/ordattmap.py", line 45, in __getitem__
    return super(OrdAttMap, self).__getitem__(item)
KeyError: 'assets'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kevin/anaconda3/envs/refgenie/bin/refgenie", line 10, in <module>
    sys.exit(main())
  File "/home/kevin/anaconda3/envs/refgenie/lib/python3.9/site-packages/refgenie/cli.py", line 241, in main
    rgc.pull(
  File "/home/kevin/anaconda3/envs/refgenie/lib/python3.9/site-packages/refgenconf/refgenconf.py", line 1485, in pull
    if not self.set_genome_alias(
  File "/home/kevin/anaconda3/envs/refgenie/lib/python3.9/site-packages/refgenconf/refgenconf.py", line 1867, in set_genome_alias
    set_aliases, removed_aliases = _check_and_set_alias(
  File "/home/kevin/anaconda3/envs/refgenie/lib/python3.9/site-packages/yacman/yacman.py", line 165, in __exit__
    self.write()
  File "/home/kevin/anaconda3/envs/refgenie/lib/python3.9/site-packages/refgenconf/refgenconf.py", line 2627, in write
    self.run_plugins(POST_UPDATE_HOOK)
  File "/home/kevin/anaconda3/envs/refgenie/lib/python3.9/site-packages/refgenconf/refgenconf.py", line 2604, in run_plugins
    func(self)
  File "/home/kevin/qbic/nfcore/refgenie_test/tools/nf_core/refgenie.py", line 82, in update_config
    refgenie_genomes = print_nf_config(rgc)
  File "/home/kevin/qbic/nfcore/refgenie_test/tools/nf_core/refgenie.py", line 33, in print_nf_config
    abg = rgc.list_assets_by_genome()
  File "/home/kevin/anaconda3/envs/refgenie/lib/python3.9/site-packages/refgenconf/refgenconf.py", line 1055, in list_assets_by_genome
    else self.list(order, include_tags=include_tags)
  File "/home/kevin/anaconda3/envs/refgenie/lib/python3.9/site-packages/refgenconf/refgenconf.py", line 333, in list
    [
  File "/home/kevin/anaconda3/envs/refgenie/lib/python3.9/site-packages/refgenconf/refgenconf.py", line 337, in <listcomp>
    list(self[CFG_GENOMES_KEY][g][CFG_ASSETS_KEY].keys()), key=order
  File "/home/kevin/anaconda3/envs/refgenie/lib/python3.9/site-packages/attmap/pathex_attmap.py", line 56, in __getitem__
    v = super(PathExAttMap, self).__getitem__(item)
  File "/home/kevin/anaconda3/envs/refgenie/lib/python3.9/site-packages/attmap/ordattmap.py", line 47, in __getitem__
    return AttMap.__getitem__(self, item)
  File "/home/kevin/anaconda3/envs/refgenie/lib/python3.9/site-packages/attmap/attmap.py", line 32, in __getitem__
    return self.__dict__[item]
KeyError: 'assets'

```

**Comment by stolarczyk** [comment]
hi! Refgenie has evolved since the creation of this plugin... It required a patch in refgenconf. Can you try using refgenconf@dev for further development? Once you confirm everything else works, we'll make a release.

**Comment by nsheff** [comment]
I can confirm @stolarczyk your fix solved the problem.

use this to test:

```
pip install --upgrade https://github.com/databio/refgenconf/archive/dev.zip
```

I also solve the problem another way in the commit in this repo (eb34056).

Basically, what was happening was, the POST_UPDATE hook runs multiple times during the PULL process; once after the genome is initialized, and then once after the asset is pulled. This is because the config file is written to both times. 

The problem is, after the first write, but before the second, when the update hook is called, the config file hadn't actually added any assets. 

In the fix to the plugin I just caught the exception to not change the nextflow config. WIth @stolarczyk's fix upstream though, this isn't necessary.

**Comment by KevinMenden** [comment]
Great, thanks both!
I will use the dev branch of `refgenconf` for now then until there is a new release :+1:

**Comment by KevinMenden** [comment]
Works like a charm now :ok_hand: :rocket:

**Comment by stolarczyk** [comment]
@KevinMenden, the newly released refgenconf version (0.11.1) includes the fix to this issue.
